specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+  /// Strict SameSite type. The cookie will only be sent along with "same-site" requests.
+  COREWEBVIEW2_COOKIE_SAME_SITE_KIND_STRICT,
+} COREWEBVIEW2_COOKIE_SAME_SITE_KIND;
+
+[uuid(20113081-93BD-4F2A-86B9-ADF92DEAAF10), object, pointer_default(unique)]
+interface ICoreWebView2_2 : ICoreWebView2 {
+  // ...
+
+  /// Gets the cookie manager object associated with this ICoreWebView2.
+  [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
+}
+
+/// Provides a set of properties that are used to manage an
+/// ICoreWebView2Cookie.
+[uuid(0AD66B3B-316F-4F54-8472-F5FF54360A60), object, pointer_default(unique)]
+interface ICoreWebView2Cookie : IUnknown {
+  /// Cookie name.
+  [propget] HRESULT Name([out, retval] LPWSTR* name);
+
+  /// Cookie value.
+  [propget] HRESULT Value([out, retval] LPWSTR* value);
+  /// Set the cookie value property.
+  [propput] HRESULT Value([in] LPCWSTR value);
+
+  /// The domain for which the cookie is valid.
+  /// The default is the host that this cookie has been received from.
+  /// Note that, for instance, ".bing.com", "bing.com", and "www.bing.com" are
+  /// considered different domains.
+  [propget] HRESULT Domain([out, retval] LPWSTR* domain);
+
+  /// The path for which the cookie is valid. The default is "/", which means
+  /// this cookie will be sent to all pages on the Domain.
+  [propget] HRESULT Path([out, retval] LPWSTR* path);
+
+  /// The expiration date and time for the cookie as the number of seconds since the UNIX epoch.
+  /// The default is -1.0, which means cookies are session cookies by default.
+  [propget] HRESULT Expires([out, retval] double* expires);
+  /// Set the Expires property. Cookies are session cookies
+  /// and will not be persistent if Expires is negative.
+  [propput] HRESULT Expires([in] double expires);
+
+  /// Whether this cookie is http-only.
+  /// True if a page script or other active content cannot access this
+  /// cookie. The default is false.
+  [propget] HRESULT IsHttpOnly([out, retval] BOOL* isHttpOnly);
+  /// Set the IsHttpOnly property.
+  [propput] HRESULT IsHttpOnly([in] BOOL isHttpOnly);
+
+  /// SameSite status of the cookie which represents the enforcement mode of the cookie.
+  /// The default is COREWEBVIEW2_COOKIE_SAME_SITE_KIND_LAX.
+  [propget] HRESULT SameSite([out, retval] COREWEBVIEW2_COOKIE_SAME_SITE_KIND* sameSite);
+  /// Set the SameSite property.
+  [propput] HRESULT SameSite([in] COREWEBVIEW2_COOKIE_SAME_SITE_KIND sameSite);
+
+  /// The security level of this cookie. True if the client is only to return
+  /// the cookie in subsequent requests if those requests use HTTPS.
+  /// The default is false.
+  /// Note that cookie that requests COREWEBVIEW2_COOKIE_SAME_SITE_KIND_NONE but
+  /// is not marked Secure will be rejected.
+  [propget] HRESULT IsSecure([out, retval] BOOL* isSecure);
+  /// Set the IsSecure property.
+  [propput] HRESULT IsSecure([in] BOOL isSecure);

```

Comment:
Something about calling a cookie "secure" feels weird. Because the thing that's secure is not the cookie, but the code that handles the cookie. ("RequiresHttps" might be a clearer name, but it looks like "Secure" is matching prior art. System.Net.Cookie.Secure boolean property.)